search-solr-importer/src/search_solr_importer/services/lear_db.py

```diff
@@ -0,0 +1,70 @@
+# Copyright © 2025 Province of British Columbia
+#
+# Licensed under the BSD 3 Clause License, (the "License");
+# you may not use this file except in compliance with the License.
+# The template for the license can be found here
+#    https://opensource.org/license/bsd-3-clause/
+#
+# Redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the
+# following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its contributors
+#    may be used to endorse or promote products derived from this software
+#    without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS “AS IS”
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Create lear db connection.
+
+These will get initialized by the application.
+"""
+from flask import Flask
+
+from search_solr_importer.services.base_db import BaseDB, DBConfig
+
+
+class LearDB:
+    """LEAR db connection."""
+
+    def __init__(self, app: Flask | None = None):
+        self.db = None
+        if app:
+            self.init_app(app)
+
+    def init_app(self, app: Flask, test_connection=True):
+        """Initialize with option to skip connection test."""
+        try:
+            db_config = DBConfig(
+                instance_connection_name=app.config["DB_CONNECTION_NAME"],
+                database=app.config["DB_NAME"],
+                user=app.config["DB_USER"],
+                enable_iam_auth=True,
+                credentials_path=app.config["GOOGLE_APPLICATION_CREDENTIALS_LEAR"],
```

Comment:
How are tested this in gcp? I would use credentials_path only if connecting to gcp db externally. From inside gcp the service account assigned to the cloud run service would auto-authenticate (if you remember I previously asked a question where you wanted a single account to have permissions to access all dbs - that would be an account assigned to your cloudrun).